src/GHC/All.hs

```diff
@@ -161,17 +161,25 @@ parseModuleEx flags file str = timedIO "Parse" file $ do
             Nothing | file == "-" -> getContentsUTF8
                     | otherwise -> readFileUTF8' file
         str <- pure $ dropPrefix "\65279" str -- remove the BOM if it exists, see #130
-        ppstr <- runCpp (cppFlags flags) file str
         let enableDisableExts = ghcExtensionsFromParseFlags flags
-        dynFlags <- parsePragmasIntoDynFlags baseDynFlags enableDisableExts file ppstr
+        dynFlags <- parsePragmasIntoDynFlags baseDynFlags enableDisableExts file str
```

Comment:
I think its important to get the pragmas after CPP. Imagine someone does:

```
{-# LANGUAGE CPP #-}
#if condition
{-# LANGUAGE QuasiQuotes #-}
#endif
```

GHC handles that.

Reply:
Ok, good point. What this means is (1) parse pragmas, (2) if CPP is enabled, do cpp and (3) parse pragmas again. I just looked into `GHC/Driver/Pipeline.hs` and indeed that's what GHC does.